api/src/main/java/com/vinumeris/updatefx/UpdateFX.java

```diff
@@ -85,9 +85,17 @@ public static Path findAppExecutable(Path appInstallDir) throws IOException {
         if (os.contains("mac")) {
             // Returns path to the .app directory, which normally contains only a Contents directory.
             List<Path> exes = Utils.listDir(appInstallDir.resolve("Contents/MacOS"));
-            if (exes.size() != 1)
-                throw new IllegalStateException("Found unknown number of app executables");
-            return exes.get(0);
+            for (Path exe : exes) {
+                /**
+                 * Since 8u40, the packager introduced a "libpackager.dylib"
```

Comment:
This is probably going to exist on other platforms too. So perhaps we should rather go with your original suggestion of ignoring any file with an extension, or perhaps matching on the "libpackager" name and ignoring the extension (which could be dll, .so etc)